### ISSUE TITLE
Installation errors

### ISSUE THREAD

**Issue by MtotheKay** [MainRo/deepspeech-server#23]
Hello,

When installing with `python3 setup.py install`, Rx 1.6.1 will be installed which is not compatible with some other components, as it seems.
`error: Rx 1.6.1 is installed but rx>=3.0 is required by {'cyclotron', 'cyclotron-std'}`

Manually increasing to 3.6 will result in an error when starting the server.py, since rx.concurrency was deprecated a long time ago, and did not make it to 3.6.

Best Regards

**Comment by MtotheKay** [comment]
The first install was by downloading the zip and executing directly. When installing though pip3 (sudo pip3 install deepspeech-server) I get the following error:

```
deepspeech-server --config config.json
Traceback (most recent call last):
  File "/usr/local/bin/deepspeech-server", line 3, in <module>
    from deepspeech_server.server import main;
  File "/usr/local/lib/python3.6/dist-packages/deepspeech_server/server.py", line 7, in <module>
    from cyclotron.router import make_error_router
  File "/usr/local/lib/python3.6/dist-packages/cyclotron-1.0.0-py3.6.egg/cyclotron/router.py", line 2, in <module>
ModuleNotFoundError: No module named 'rx.operators'
```
Am I missing something?

**Comment by ModelWorldLLC** [comment]
Edit: My new issue was unrelated and fixed. I can confirm that this problem exists (at least for python 3.7 on Windows), and that it can be fixed by installing Rx 1.6.1, cyclotron 0.6.1, and cyclotron-std 0.5.0. 

I have run into the same issue. I tried to resolve by downgrading cyclotron and cyclotron-std to versions compatible with Rx 1.6.1. This appears to work, until I try to actually start the server, at which point I get a cascade of errors with rx/cyclotron that I'm still trying to unravel, and which may or may not be related.

**Comment by MainRo** [comment]
The dependencies has been updated but I did not upgraded deepspeech-server yet.
In the meantime, downgrading cyclotron and cyclotron-std to <1.0 should work.

**Comment by ghost** [comment]
```
sudo pip3 install "cyclotron<1.0" "cyclotron-std<1.0"

Collecting cyclotron-std<1.0
  Downloading https://www.piwheels.org/simple/cyclotron-std/cyclotron_std-0.5.0-py3-none-any.whl
Requirement already satisfied: rx<2.0,>=1.6 in /usr/local/lib/python3.7/dist-packages (from cyclotron<1.0) (1.6.1)
Installing collected packages: cyclotron-std
  Found existing installation: cyclotron-std 1.0.0
    Uninstalling cyclotron-std-1.0.0:
      Successfully uninstalled cyclotron-std-1.0.0
Successfully installed cyclotron-std-0.5.0
pi@raspberrypi:~ $ deepspeech-server
Traceback (most recent call last):
  File "/usr/local/bin/deepspeech-server", line 3, in <module>
    from deepspeech_server.server import main;
  File "/usr/local/lib/python3.7/dist-packages/deepspeech_server/server.py", line 4, in <module>
    from rx.concurrency import AsyncIOScheduler
ModuleNotFoundError: No module named 'rx.concurrency'
```

**Comment by MtotheKay** [comment]
Hello,

I Installed cyclotron 0.6.1, cyclotron-std 0.5.0 and rx 1.6.1, by hand after installing the server. And I'm getting the same error as voltagex.

**Comment by MtotheKay** [comment]
Let's add some details:
OS: Ubuntu 18.04

Steps to duplicate:

```
pip3 install deepspeech
pip3 install deepspeech-server
pip3 uninstall cyclotron
pip3 uninstall cyclotron-std
pip3 install "cyclotron==0.6.1"
pip3 install "cyclotron-std==0.5.0"
pip3 uninstall rx
pip3 install "rx==1.6.1"

deepspeech-server --config config.json

Traceback (most recent call last):
  File "/home/username/.local/bin/deepspeech-server", line 3, in <module>
    from deepspeech_server.server import main;
  File "/home/username/.local/lib/python3.6/site-packages/deepspeech_server/server.py", line 7, in <module>
    from cyclotron.router import make_error_router
  File "/home/username/.local/lib/python3.6/site-packages/cyclotron/router.py", line 2, in <module>
    import rx.operators as ops
ModuleNotFoundError: No module named 'rx.operators'
```

**Comment by krupalkp97** [comment]
Hello ,
i faced the same error and with exact version of cyclotron==0.6.1 ,cyclotron-std==0.5.0 and rx==1.6.1 it is running. see pip3 list for version info of rx and above modules once

**Comment by MtotheKay** [comment]
Hello @krupalkp97,

thanks for the tip. Somehow cyclotron 1.0.0 was installed instead of 0.6.1. 

So  the an error Message changed:
```
~$deepspeech-server --config config.json
Illegal instruction (core dumped)
```

I can't seem to find a log file, even with the "log": section enabled in the config.json

Did you have that as well?

**Comment by MainRo** [comment]
This works for me, from a clean virtualenv:

```
pip install "Rx<1.6.1"
pip install "cyclotron<1.0"
pip install "cyclotron-std<1.0"
pip install "cyclotron-aio<1.0"
pip install deepspeech-server
```

Do you use deepspeech 0.5.1 or one of the 0.6 alphas ?

**Comment by MtotheKay** [comment]
Hi @MainRo,

aiohttp >3 only runs with python > 3, so I guess you used pip3?
Also cyclotron-std<1.0 installs cyclotron-1.0.0 which in return installs Rx-1.6.1, so I started with deepspeech-server, then cyclotron-aio, then cyclotron-std, then cyclotron and finaly Rx. (Uninstalling before installing)

This the relevat pip3 list output 
```
cyclotron (0.6.1)
cyclotron-aio (0.7.0)
cyclotron-std (0.5.0)
deepspeech (0.5.1)
deepspeech-server (1.1.0)
Rx (1.6.0)
```

here the complete list:

```
acme (0.31.0)
aiohttp (3.6.2)
apturl (0.5.2)
asn1crypto (0.24.0)
async-timeout (3.0.1)
attrs (19.3.0)
Brlapi (0.6.6)
certbot (0.27.0)
certifi (2018.1.18)
chardet (3.0.4)
command-not-found (0.3)
ConfigArgParse (0.11.0)
configobj (5.0.6)
cryptography (2.1.4)
cupshelpers (1.0)
cyclotron (0.6.1)
cyclotron-aio (0.7.0)
cyclotron-std (0.5.0)
deepspeech (0.5.1)
deepspeech-server (1.1.0)
defer (1.0.6)
distro-info (0.18ubuntu0.18.04.1)
future (0.15.2)
httplib2 (0.9.2)
idna (2.8)
idna-ssl (1.1.0)
josepy (1.1.0)
keyring (10.6.0)
keyrings.alt (3.0)
language-selector (0.1)
launchpadlib (1.10.6)
lazr.restfulclient (0.13.5)
lazr.uri (1.0.3)
louis (3.5.0)
macaroonbakery (1.1.3)
mock (2.0.0)
multidict (4.6.1)
netifaces (0.10.4)
numpy (1.17.4)
oauth (1.0.1)
olefile (0.45.1)
opencv-python (4.1.1.26)
parsedatetime (2.4)
pbr (3.1.1)
pexpect (4.2.1)
Pillow (5.1.0)
pip (9.0.1)
protobuf (3.0.0)
pycairo (1.16.2)
pycrypto (2.6.1)
pycryptodomex (3.6.1)
pycups (1.9.73)
pygobject (3.26.1)
pymacaroons (0.13.0)
PyNaCl (1.1.2)
pyOpenSSL (17.5.0)
pyRFC3339 (1.0)
python-apt (1.6.4)
python-dateutil (2.6.1)
python-debian (0.1.32)
pytz (2018.3)
pyxdg (0.25)
PyYAML (3.12)
reportlab (3.4.0)
requests (2.18.4)
requests-toolbelt (0.8.0)
requests-unixsocket (0.1.5)
Rx (1.6.0)
scipy (1.3.3)
SecretStorage (2.3.1)
setuptools (39.0.1)
simplejson (3.13.2)
six (1.11.0)
ssh-import-id (5.7)
system-service (0.3)
systemd-python (234)
typing-extensions (3.7.4.1)
ubuntu-drivers-common (0.0.0)
ufw (0.36)
unattended-upgrades (0.1)
urllib3 (1.22)
wadllib (1.3.2)
wheel (0.30.0)
xkit (0.0.0)
yarl (1.4.1)
zope.component (4.3.0)
zope.event (4.2.0)
zope.hookable (4.0.4)
zope.interface (4.3.2)
```
I tried Version 0.5.1 and 0.6.0 of deepspeech.

Does anybody know where to find the logs?

**Comment by MainRo** [comment]
yes, I use pip3, but form a virtualenv. So it is named pip in my environment.
You list seems correct to me. Are you able to use the native client of deepspeech ? The coredump may be related to another issue with deepspeech.
The logs are supposed to be printed on the console, but the program crashes before anything is printed.

**Comment by kirill-dev-pro** [comment]
after install in a clean virtualenv 
```pkg_resources.ContextualVersionConflict: (Rx 1.6.1 (/usr/local/lib/python3.7/dist-packages/Rx-1.6.1-py3.7.egg), Requirement.parse('rx>=3.0'), {'cyclotron', 'cyclotron-std'})```
python v3.7.5
pip v19.3.1

**Comment by MainRo** [comment]
did you try to manually install the dependencies before deepspeech-server, as explained above ? https://github.com/MainRo/deepspeech-server/issues/23#issuecomment-560631462

**Comment by kirill-dev-pro** [comment]
Yes, I tried instalation through requirements.txt, manual instalation and instalation with requirements.txt with manually installed rx, cyclotron and cyclotron-std

**Comment by MainRo** [comment]
what is your output of pip freeze ?
I am currently porting to rxpy v3, so these installation issues will be fixed soon.

**Comment by MainRo** [comment]
This is fixed in v2.0.0
